R/fwrite.R

```diff
@@ -101,14 +104,22 @@ fwrite <- function(x, file="", append=FALSE, quote="auto",
         header = col.names, sep = sep, sep2 = sep2, eol = eol, na.strings = na,
         dec = dec, qmethod = qmethod, logical01 = logical01
       )
+      if (with_bom) {
+          bom <- raw(3L)
+          bom[1] <- as.raw(0xEF)
+          bom[2] <- as.raw(0xBB)
+          bom[3] <- as.raw(0xBF)
+          writeBin(bom, file)
```

Comment:
Does this interact properly with `append`? I guess `with_bom && append` should be another error or ignore case

Reply:
I see, this is nested within `if(yaml)` so already `append` case is handled

Reply:
I would add internal `stopifnot` just in case if the code will be moved later on, to ensure what Michael mentioned.